### ISSUE TITLE
Umbrella cover crashes when no sub-app name is specified.

### ISSUE THREAD

**Issue by hickscorp** [parroty/excoveralls#142]
Excoveralls crashes with the following error if there is no `:app_name` key in the mix config for umbrella sub-apps:

**EDIT: Scratch that - even having `app_name` doesn't help at all.**

```
** (FunctionClauseError) no function clause matching in anonymous fn/1 in ExCoveralls.Stats.append_sub_app_name/3

    lib/excoveralls/stats.ex:93: anonymous fn/1 in ExCoveralls.Stats.append_sub_app_name/3
    (elixir) lib/enum.ex:1294: Enum."-map/2-lists^map/1-0-"/2
    lib/excoveralls.ex:49: ExCoveralls.store_stats/3
    (mix) lib/mix/tasks/test.ex:273: Mix.Tasks.Test.run/1
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/project.ex:351: Mix.Project.in_project/4
    (elixir) lib/file.ex:1419: File.cd!/2
    (mix) lib/mix/task.ex:414: anonymous fn/4 in Mix.Task.recur/1
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mix) lib/mix/task.ex:413: Mix.Task.recur/1
    (mix) lib/mix/project_stack.ex:141: Mix.ProjectStack.recur/1
    (mix) lib/mix/task.ex:348: Mix.Task.run_alias/3
    (mix) lib/mix/task.ex:277: Mix.Task.run/2
    lib/mix/tasks.ex:51: Mix.Tasks.Coveralls.do_run/2
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:80: Mix.CLI.run_task/2
    (elixir) lib/code.ex:677: Code.require_file/2
```

It might seem obvious for some of us, but it surprised me as a project was using 0.8.2 and it started failing our CI when we upgraded to 0.9.0.

**Comment by parroty** [comment]
Thanks for the report. I've merged the #141 and pushed as v0.9.1. Does this resolve the issue?

**Comment by hickscorp** [comment]
@parroty it worked indeed, thanks. I'll be closing this issue then.
